### PR TITLE
The background system for shop

### PR DESCRIPTION
**Pull request by OtakuFreak13**
The shop has all the functions needed for making a shop. Only the UI left, which will be done in the next task when 2DUI is finished.

For testing in Andres test scene in sandbox I have filled the shop inventory and then applied both the test upgrades. Since they no longer print stuff the only way to see if they are applied is by shooting a projectile at a barbarian to see if they start flying upwards .

### CODE REVIEW

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h [hunk 1]
```diff
@@ -63,6 +65,10 @@ class Upgrade
 	Entity* m_pParentEntity;
 	// Name of the upgrade, for ease of access in shop or upgrade handlers
 	std::string m_Name = "";
+	// Discription is used in shop when describing the upgrade.
+	std::string m_Discription;
```
Comment (Schtekt) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h]: Shouldn't the variable be description?

**Review by Whixx — DISMISSED** [review]
Nice

**Review by Schtekt — CHANGES_REQUESTED** [review]
Just change that variable and then it looks good :)

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h (hunk 1 above)
Comment (OtakuFreak13) [reply StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h]: Done

File: StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h [hunk 2]
```diff
@@ -63,6 +65,10 @@ class Upgrade
 	Entity* m_pParentEntity;
 	// Name of the upgrade, for ease of access in shop or upgrade handlers
 	std::string m_Name = "";
+	// Discription is used in shop when describing the upgrade.
```
Comment (Whixx) [inline StortSpelProjekt/Game/src/Gamefiles/Components/UpgradeComponents/Upgrades/Upgrade.h]: dIIIIIII

File: StortSpelProjekt/Game/src/Gamefiles/Shop.h [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	// Change player currency balance
+	void ChangePlayerBalance(int change);
+
+	// Get size of inventory
+	int GetInventorySize();
+	// Get vector of buyable upgrades names.
+	std::vector<std::string> GetInventoryNames();
+	// Get descriptions, for upgrades in inventory, mapped to their names.
+	std::map<std::string, std::string> GetUpgradeDiscriptions();
+	// Get discription for an upgrade in inventory using its name as input.
```
Comment (Whixx) [inline StortSpelProjekt/Game/src/Gamefiles/Shop.h]: d I scription

**Review by Whixx — CHANGES_REQUESTED** [review]

**Review by Whixx — DISMISSED** [review]

**Review by Whixx — APPROVED** [review]

**Review by Schtekt — APPROVED** [review]

**Pull request merged**